Runtime/Controller/JointControl.cs

```diff
@@ -39,9 +39,14 @@ void Start()
             {
                 ArticulationDrive currentDrive = joint.xDrive;
                 float newTargetDelta = (int)direction * Time.fixedDeltaTime * speed;
-                if (newTargetDelta + currentDrive.target <= currentDrive.upperLimit && newTargetDelta + currentDrive.target >= currentDrive.lowerLimit){
+                if (((joint.linearLockX == ArticulationDofLock.LimitedMotion) && (newTargetDelta + currentDrive.target <= currentDrive.upperLimit && newTargetDelta + currentDrive.target >= currentDrive.lowerLimit)) || 
```

Comment:
It seems the target changes are the same no matter what the condition is. Could you double-check?

Reply:
Yes they are the same, I could have created a single statement but this is more for clarity. In the first case, it checks for wether the joint is limited and than if the target is within the limits. In the second one it lets it set the target.